### DISCUSSION TITLE
Languages parsed incorrectly on search and repository page

### DISCUSSION THREAD

**Discussion by mishushakov** [community/community#36562] [Repositories]
See this: https://github.com/stepci/stepci/search?l=javascript

**Comment by spenserblack** [comment]
It is likely the shebang `#!/usr/bin/env node` that is causing it to be classified as JS.
You can use [gitattributes overrides](https://github.com/github/linguist/blob/master/docs/overrides.md#using-gitattributes) to force language detection. For example:
```gitattributes
*.ts linguist-language=TypeScript
```

Personally, instead of putting a shebang in the Typescript file itself, I usually do something like this:
```javascript
#!/usr/bin/env node
// bin/foo.js
require('../dist/foo');
```
```typescript
// src/foo.ts
console.log('Hello, World!');
```
which should also result in the correct classification of languages.

**Comment by mishushakov** [comment]
I don't think this is the case here. It appears that GitHub won't reindex the repository. When I click on "JavaScript" it shows me the old `src/index.ts` file and says "Last indexed 12 days ago". But, my latest commit was like 20 minutes ago!

**Reply by spenserblack** [reply]
It takes a while for the index to update, but check out [this repo](https://github.com/spenserblack/test-repo/search?l=javascript). I've gotten a Python file to be considered JavaScript using the `#!/usr/bin/env node` shebang.

**Reply by mishushakov** [reply]
Yep, that's clearly not how you would expect things to work

**Reply by spenserblack** [reply]
And for fun I also have a CSV classified as JavaScript :laughing:

**Reply by mishushakov** [reply]
JavaScript runs the world. Soon enough anything will be JavaScript

**Comment by ossanna16** [comment]
Hey there @mishushakov and thank you for the feedback! I'm glad you found our community 🙂

**Comment by spenserblack** [comment]
So it looks like I was mistaken in my initial answer (https://github.com/community/community/discussions/36562#discussioncomment-3906937) about `.gitattributes`. github-linguist will use that, but it looks like GitHub's search tool uses [go-enry](https://github.com/go-enry/go-enry), which currently doesn't support `.gitattributes` (pending issue: https://github.com/go-enry/go-enry/issues/27). I'm guessing that shebangs take higher priority over filenames in GitHub's usage.
It looks like a vim modeline takes higher priority over a shebang, so you could use it to force the search tool to classify the file correctly.
```typescript
#!/usr/bin/env node
// vim: set ft=typescript:
// rest of the file...
```

https://github.com/spenserblack/test-repo/search?l=typescript
